### DISCUSSION TITLE
How to set default value of a field using the Postgresql db operators or otherwise

### DISCUSSION THREAD

**Discussion by palashCItobuz** [prisma/prisma#9054] [Q&A] [answered]
I need to add a `fiscal_year` to the schema and when I add something this it shows.. `Error validating: This line is not a valid field or attribute definition.`
```prisma
fiscalYear             Int         @default(new Date('Y')) @map("fiscal_year")
```
OR 
How can I use the database functions such as `CURRENT_TIMESTAMP` to derive the current year? 

```prisma
fiscalYear             DateTime             @default(date_part('year', now())) @map("fiscal_year")
```

**Comment by ryands17** [comment, marked as answer]
@palashCItobuz 👋 

You can use [`@dbgenerated`](https://www.prisma.io/docs/reference/api-reference/prisma-schema-reference#override-default-value-behavior-for-supported-types) to achieve this.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
